gateway/gateway_messages.go

```diff
@@ -437,57 +436,77 @@ type IdentifyCommandDataProperties struct {
 	Device  string `json:"device"`  // library name
 }
 
-// NewPresence creates a new Presence with the provided properties
-func NewPresence(activityType discord.ActivityType, name string, url string, status discord.OnlineStatus, afk bool) MessageDataPresenceUpdate {
-	var since *int64
-	if status == discord.OnlineStatusIdle {
-		unix := time.Now().Unix()
-		since = &unix
-	}
+type PresenceOpt func(presenceUpdate *MessageDataPresenceUpdate)
 
-	var activities []discord.Activity
-	if name != "" {
-		activity := discord.Activity{
-			Name: name,
-			Type: activityType,
-		}
-		if activityType == discord.ActivityTypeStreaming && url != "" {
-			activity.URL = &url
-		}
-		activities = append(activities, activity)
-	}
+// WithPlayingActivity creates a new "Playing ..." activity of type discord.ActivityTypeGame
+func WithPlayingActivity(name string) PresenceOpt {
```

Comment:
should we rename those functions to `WithActivityX` for better discoverability?

Reply:
I think it's safe to assume people these days use intelligent IDEs which will already suggest the methods when using proper keywords such as "activity" or "listening".